frontend/app/src/App.test.tsx

```diff
@@ -219,1619 +369,1738 @@ describe("App", () => {
+      //   3. Script had a compilation error
+      // The other solutions test the following:
+      //   1. The script's initial state of not running
+      //   2. A script rerun was requested
+      //   3. A script stop was requested
 
-    // Send close message
-    dispatchEvent(
-      "message",
-      new MessageEvent("message", {
-        data: {
-          stCommVersion: HOST_COMM_VERSION,
-          type: "CLOSE_MODAL",
-        },
-        origin: "https://devel.streamlit.test",
+      const hostCommunicationMgr = prepareHostCommunicationManager()
+
+      sendForwardMessage("sessionStatusChanged", {
+        runOnSave: false,
+        scriptIsRunning: true,
       })
-    )
-    expect(wrapper.find(Modal)).toHaveLength(0)
 
-    // Open a dialog again - make sure it works
-    wrapper.setState({ dialog })
-    expect(wrapper.find(Modal)).toHaveLength(1)
-  })
+      expect(hostCommunicationMgr.sendMessageToHost).toHaveBeenCalledWith({
+        type: "SCRIPT_RUN_STATE_CHANGED",
+        scriptRunState: ScriptRunState.RUNNING,
+      })
 
-  it("sends SCRIPT_RUN_STATE_CHANGED signal to the host when scriptRunState changing", () => {
-    Object.keys(ScriptRunState).forEach(scriptRunState => {
-      wrapper.setState({ scriptRunState })
-      expect(sendMessageFunc).toHaveBeenCalledWith({
+      sendForwardMessage("sessionStatusChanged", {
+        runOnSave: false,
+        scriptIsRunning: false,
+      })
+
+      expect(hostCommunicationMgr.sendMessageToHost).toHaveBeenCalledWith({
         type: "SCRIPT_RUN_STATE_CHANGED",
-        scriptRunState,
+        scriptRunState: ScriptRunState.NOT_RUNNING,
+      })
+
+      sendForwardMessage("sessionEvent", {
+        type: "scriptCompilationException",
+      })
+
+      expect(hostCommunicationMgr.sendMessageToHost).toHaveBeenCalledWith({
+        type: "SCRIPT_RUN_STATE_CHANGED",
+        scriptRunState: ScriptRunState.COMPILATION_ERROR,
       })
     })
-  })
 
-  it("does not sends SCRIPT_RUN_STATE_CHANGED signal to the host when scriptRunState changing to the same state", () => {
-    const scriptRunState = ScriptRunState.RUNNING
-    wrapper.setState({ scriptRunState })
-    expect(sendMessageFunc).toHaveBeenCalledWith({
-      type: "SCRIPT_RUN_STATE_CHANGED",
-      scriptRunState: scriptRunState,
-    })
-    expect(sendMessageFunc).toHaveBeenCalledTimes(1)
-    // When scriptRunState changed to the same,
-    // sendMessage should not be called again.
-    wrapper.setState({ scriptRunState })
-    expect(sendMessageFunc).toHaveBeenCalledTimes(1)
-  })
+    it("does not sends SCRIPT_RUN_STATE_CHANGED signal to the host when scriptRunState changing to the same state", () => {
+      const hostCommunicationMgr = prepareHostCommunicationManager()
+
+      sendForwardMessage("sessionStatusChanged", {
+        runOnSave: false,
+        scriptIsRunning: true,
+      })
 
-  it("both sets theme locally and sends to host when setAndSendTheme is called", () => {
```

Comment:
Think we should probably still have some test for the `SET_THEME_CONFIG` host message

Reply:
I think this is in the test labeled "sends theme info to the host when the app is first rendered", but let me know if I am missing something.